Formula/osquery.rb

```diff
@@ -80,6 +79,14 @@ class Osquery < Formula
     sha256 "61955afa09ef244fc84a72ef019de15515e76377aceeb2cbf1e93fa0df374cd2"
   end
 
+  # Remove for > 2.3.2
+  # Fix "error: 'shared_timed_mutex' is unavailable"
+  # Upstream commit from 15 Feb 2017 "darwin: Use boost shared_mutex for OS X"
+  patch do
+    url "https://github.com/facebook/osquery/commit/d1d21cd.patch"
```

Comment:
Ah, for macOS this is essentially 2.3.3, so you can pull in the patch or use the pre-release tag, either way is fine.